documentation/library_guide/parallel_api/pass_data_algorithms.rst

```diff
@@ -81,47 +76,65 @@ the buffer were created for the same queue. For example:
     return 0;
   }
 
-Alternatively, use ``std::vector`` with a USM allocator. For example:
+When using device USM, such as allocated by ``malloc_device``, manually copy data to this memory
+before calling oneDPL algorithms, and copy it back once the algorithms have finished execution.
+
+Use std::vector
+-----------------------------
+
+The following examples demonstrate two ways to use the parallel algorithms with ``std::vector``:
+
+* host allocators
+* USM allocators
+
+You can use iterators to host allocated ``std::vector`` data
+as shown in the following example:
 
 .. code:: cpp
 
   #include <oneapi/dpl/execution>
   #include <oneapi/dpl/algorithm>
-  #include <sycl/sycl.hpp>
+  #include <vector>
   int main(){
-    const int n = 1000;
-    auto policy = oneapi::dpl::execution::dpcpp_default;
-    sycl::usm_allocator<int, sycl::usm::alloc::shared> alloc(policy.queue());
-    std::vector<int, decltype(alloc)> vec(n, alloc);
-
-    std::fill(policy, vec.begin(), vec.end(), 42);
-
+    std::vector<int> vec( 1000 );
+    std::fill(oneapi::dpl::execution::dpcpp_default, vec.begin(), vec.end(), 42);
+    // each element of vec equals to 42
     return 0;
   }
 
-When using device USM, such as allocated by ``malloc_device``, manually copy data to this memory
-before calling oneDPL algorithms, and copy it back once the algorithms have finished execution.
-
-Use Host-Side std::vector
------------------------------
+When using iterators to host allocated data, a temporary SYCL buffer is created, and the data
+is copied to this buffer. After processing on a device is complete, the modified data is copied
+from the temporary buffer back to the host container. While convenient, using host allocated
+data can lead to unintended copying between host and device. We recommend working with SYCL buffers
+or USM memory to reduce data copying between the host and device. 
 
-|onedpl_long| parallel algorithms can be called with ordinary (host-side) iterators, as seen in the
-example below.
-In this case, a temporary SYCL buffer is created, and the data is copied to this buffer.
-After processing on a device is complete, the modified data is copied from the temporary buffer back
-to the host container.
-For example:
+You can also use ``std::vector`` with a USM allocator, as shown in the following example:
 
 .. code:: cpp
 
   #include <oneapi/dpl/execution>
   #include <oneapi/dpl/algorithm>
-  #include <vector>
+  #include <sycl/sycl.hpp>
```

Comment:
I believe we don't want to have underscored lines in the code examples.
![code_example](https://github.com/oneapi-src/oneDPL/assets/40659946/0c465027-09f5-4e0d-830c-72f8db015ec8)

Reply:
I think that is just a rendering quirk for github and rst.  I don't think we are doing anything to have the code underlined.  Perhaps @dcbenito  can confirm.

Reply:
I ran a test build of the documentation and there are no underlines there. I am confirming this is a rendering quirk.